src/core_landice/mode_forward/mpas_li_velocity.F

```diff
@@ -916,6 +963,35 @@ subroutine calculate_strain_rates(meshPool, velocityPool, err)
       eMax = 0.5_RKIND * (exx + eyy) + sqrt( (0.5_RKIND * (exx - eyy))**2 + (0.25_RKIND*(exy + eyx))**2)
       eMin = 0.5_RKIND * (exx + eyy) - sqrt( (0.5_RKIND * (exx - eyy))**2 + (0.25_RKIND*(exy + eyx))**2)
 
+      call li_calculate_flowParamA(meshPool, temperature, thickness, flowParamA, err_tmp)
+      err = ior(err, err_tmp)
+
+     ! calculate the depth-averaged flow parameter A
+      meanFlowParamA(:) = sum(flowParamA(:,:), dim=1)/REAL(nVertLevels)      
+
+      ! calculate effective viscosity from strain rate and flow param.
+      where (thickness == 0.0_RKIND)
+          effectiveViscosity = 0.0_RKIND
+      elsewhere
+          effectiveViscosity = &
+          0.5_RKIND*stiffnessFactor*meanFlowParamA**(-1.0_RKIND/3.0_RKIND)*(sqrt(exx**2.0_RKIND+eyy**2.0_RKIND + &
+          exx*eyy + 0.25_RKIND*(exy+eyx)**2.0_RKIND))**((1.0_RKIND-3.0_RKIND)/3.0_RKIND)
+      endwhere
+
+      ! calculate deviatoric stresses
+      sigmaxx(:) = 2.0_RKIND*effectiveViscosity(:)*exx(:)
+      sigmayy(:) = 2.0_RKIND*effectiveViscosity(:)*eyy(:)
+      sigmaxy(:) = 2.0_RKIND*effectiveViscosity(:)*(exy(:)+eyx(:))/2.0_RKIND
+      sigmayx(:) = 2.0_RKIND*effectiveViscosity(:)*(eyx(:)+exy(:))/2.0_RKIND
+
+      ! calculate principal stresses
+      sigmamin(:) = &
+      (sigmaxx(:)+sigmayy(:))/2.0_RKIND-sqrt(((sigmaxx(:)-sigmayy(:))/2.0_RKIND)**2.0_RKIND+((sigmaxy(:)+sigmayx(:))/2.0_RKIND)**2.0_RKIND)
```

Comment:
I don't know any reason why the equations for deviatoric principal stresses should be different than for full principal stresses. @tongzhangice should comment though. I don't know if I've ever derived that from scratch either.

Reply:
@tongzhangice -- we believe that the first term in both the min and max principal stress equations ... the (sigmaxx(:)+sigmayy(:))/2.0_RKIND  would look like this in the case of using deviatoric stresses:

(tauxx(:)+tauyy(:) + 2*P ) / 2.0_RKIND + ...

where P is the pressure. No other terms are affected (the pressures would cancel in one of the other terms and the shear stress terms are the same for dev. and full stresses).

For a depth-varying calculation, this would be P(z). At the sfc, this would not change the current calculation. At the bed, this term would be 2 * rho*g*H, and if we were using depth-averaged strain rates for the dev. stresses, I would be 2 * 1/2 * rho * g * H.

Can you confirm if this is what you were using in your analysis scripts to calc. principal stresses for the TC paper on ice shelf sensitivity to basal melting?